pkg/provisioner/ironic/ironic.go

```diff
@@ -508,23 +508,30 @@ func (p *ironicProvisioner) InspectHardware() (result provisioner.Result, detail
 				result.RequeueAfter = introspectionRequeueDelay
 				err = nil
 				return
+			case nodes.Manageable:
+				inspect = true
 			default:
-				p.log.Info("starting new hardware inspection")
-				var success bool
-				success, result, err = p.tryChangeNodeProvisionState(
-					ironicNode,
-					nodes.ProvisionStateOpts{Target: nodes.TargetInspect},
-				)
-				if success {
-					p.publisher("InspectionStarted", "Hardware inspection started")
-				}
-				return
+				inspect = true
 			}
+		} else {
+			err = errors.Wrap(err, "failed to extract hardware inspection status")
+			return
+		}
+	}
+	if reinstrospect && !(nodes.ProvisionState(ironicNode.ProvisionState) == nodes.Inspecting || nodes.ProvisionState(ironicNode.ProvisionState) == nodes.InspectWait) {
```

Comment:
Typo in 'reintrospect' and also this is not defined with any spelling

A switch on the ProvisionState might be cleaner.

Reply:
Fixed, thanks.